app/src/main/java/com/github/studydistractor/sdp/user/FirebaseUserService.kt

```diff
@@ -0,0 +1,38 @@
+package com.github.studydistractor.sdp.user
+
+import android.util.Log
+import androidx.compose.runtime.mutableStateListOf
+import com.google.firebase.database.*
+import javax.inject.Inject
+
+class FirebaseUserService @Inject constructor(): UserService {
+    private val path = "Users"
+    private val databaseRef: DatabaseReference = FirebaseDatabase.getInstance().getReference(path)
+
+    override fun fetchUsers(callback: (List<UserData>) -> Unit) {
+        val result = mutableStateListOf<UserData>()
```

Comment:
I know I made the same mistake but now we need to add the listener in init { } and put the val result as attribute of the class. and the fetch function only returns the reference to result. This way the result observe the changes only once and the result is updated only when Firebase tells it to.